### DISCUSSION TITLE
Unique field validation while updating in adonis 5

### DISCUSSION THREAD

**Discussion by subhamcodes** [adonisjs/core#1288] [Help] [answered]
"@adonisjs/core": "^5.0.0-preview-rc"

There is no description regarding the unique rule. I want to check for unique fields while updating.

    schema.create({
        email: schema.string({}, [
            rules.email(),
            rules.unique({ table: 'users', column: 'email' })
        ])
    })

**Comment by ammezie** [comment, marked as answer]
You can check the docs for the `exists` rule: https://preview.adonisjs.com/guides/validator/rules#rulesexists

Same can be applied to the `unique` rule.

**Comment by dxsuporte** [comment]
```
	public refs = schema.refs({
		id: this.ctx.params.id
	})

	public schema = schema.create({

		permission_name: schema.string({ trim: true }, [
			rules.unique({ table: 'permissions', column: 'permission_name', whereNot: { id: this.refs.id } }),
			rules.minLength(3),
		]),
		active: schema.string({ trim: true })
	})
```

**Reply by salvamatavele** [reply]
{ I like this way :  this.perfect }

**Reply by bhargavfortius** [reply]
> ```
> 	public refs = schema.refs({
> 		id: this.ctx.params.id
> 	})
> 
> 	public schema = schema.create({
> 
> 		permission_name: schema.string({ trim: true }, [
> 			rules.unique({ table: 'permissions', column: 'permission_name', whereNot: { id: this.refs.id } }),
> 			rules.minLength(3),
> 		]),
> 		active: schema.string({ trim: true })
> 	})
> ```

hello sir @ammezie 
this way not working properly
